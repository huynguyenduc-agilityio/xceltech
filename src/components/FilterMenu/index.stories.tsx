import type { Meta, StoryObj } from '@storybook/react';

// Components
import FilterMenu from '.';

const defaultProps = {
  options: ['Name', 'Type'],
  onApply: () => {},
};

const meta = {
  title: 'Components/FilterMenu',
  component: FilterMenu,
  parameters: {
    layout: 'centered',
  },

  tags: ['autodocs'],
  render: (args) => <FilterMenu {...args} />,
} satisfies Meta<typeof FilterMenu>;

export default meta;

type Story = StoryObj<typeof FilterMenu>;

export const Default: Story = {
  args: { ...defaultProps },
};
