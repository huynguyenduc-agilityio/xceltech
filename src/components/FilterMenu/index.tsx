import { useState } from 'react';

// Icons
import { FilterIcon } from '@/icons';

// Components
import { Button, Checkbox, PopoverContainer } from '../common';

export interface FilterMenuProps {
  options: string[];
  selectedValues?: string[];
  onApply: (values: string[]) => void;
}

const FilterMenu = ({
  options,
  selectedValues = [],
  onApply,
}: FilterMenuProps) => {
  const [selected, setSelected] = useState<string[]>(selectedValues);

  const toggleSelection = (value: string) => {
    setSelected((prev) =>
      prev.includes(value)
        ? prev.filter((item) => item !== value)
        : [...prev, value],
    );
  };

  const resetFilters = () => setSelected([]);

  const handleFilter = () => {
    onApply(selected);
  };

  return (
    <PopoverContainer
      trigger={
        <Button variant="ghost" size="icon" className="p-2">
          <FilterIcon />
        </Button>
      }
      contentClassName="px-0 pt-4 pb-2"
      content={
        <div>
          <div className="space-y-2 px-4">
            {options.map((option) => (
              <div key={option} className="flex items-center space-x-2">
                <Checkbox
                  id={option}
                  checked={selected.includes(option)}
                  onCheckedChange={() => toggleSelection(option)}
                />
                <label htmlFor={option} className="text-sm">
                  {option}
                </label>
              </div>
            ))}
          </div>
          <div className="mt-4 border-t border-gray-muted">
            <div className="flex justify-between px-4 pt-2">
              <Button
                variant="outline"
                onClick={resetFilters}
                className="text-sm px-4 py-2 w-[60px] h-[36px] rounded-md"
              >
                Reset
              </Button>
              <Button
                className="text-sm px-4 py-2 w-[60px] h-[36px] rounded-md"
                onClick={handleFilter}
              >
                OK
              </Button>
            </div>
          </div>
        </div>
      }
    />
  );
};

export default FilterMenu;
