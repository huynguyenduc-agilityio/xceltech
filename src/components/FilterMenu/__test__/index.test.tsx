import { render } from '@testing-library/react';

// Components
import FilterMenu from '..';

describe('FilterMenu component', () => {
  const mockProps = {
    options: ['Name', 'Type'],
    onApply: jest.fn(),
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<FilterMenu {...mockProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
