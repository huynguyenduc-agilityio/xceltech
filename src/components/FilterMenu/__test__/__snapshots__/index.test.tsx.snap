// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu component should match snapshot 1`] = `
<DocumentFragment>
  <button
    aria-controls="radix-:r0:"
    aria-expanded="false"
    aria-haspopup="dialog"
    class=""
    data-state="closed"
    type="button"
  />
  <button
    class="font-productsans inline-flex items-center justify-center whitespace-nowrap text-lg font-bold ring-offset-background transition-all duration-300 ease-in-out outline-none focus-visible:outline-none disabled:opacity-50 disabled:cursor-not-allowed hover:bg-accent hover:text-accent-foreground h-10 w-10 rounded-full p-2"
  >
    <svg
      fill="none"
      height="25"
      viewBox="0 0 31 25"
      width="31"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#a)"
      >
        <path
          d="M29.545 0H1.455C.165 0-.487 1.263.427 2l11.198 9.032v10.062c0 .382.231.74.62.96l4.844 2.733c.955.54 2.286-.007 2.286-.96V11.032l11.198-9.031C31.486 1.264 30.839 0 29.545 0"
          fill="#1A1A1A"
        />
      </g>
      <defs>
        <clippath
          id="a"
        >
          <path
            d="M0 0h31v25H0z"
            fill="#fff"
          />
        </clippath>
      </defs>
    </svg>
  </button>
</DocumentFragment>
`;
